File: java/CoolingSystem.java

public class CoolingSystem extends Komponen {
    private String tipePendingin;
    private int kecepatanRpms;

    public CoolingSystem() {
        super();
        this.tipePendingin = "";
        this.kecepatanRpms = 0;
    }

    public CoolingSystem(String merk, String nama, String tipePendingin, int kecepatanRpms) {
        super(merk, nama);
        this.tipePendingin = tipePendingin;
        this.kecepatanRpms = kecepatanRpms;
    }

    // Getter
    public String getTipePendingin() {
        return tipePendingin;
    }

    public int getKecepatanRpms() {
        return kecepatanRpms;
    }

    // Setter
    public void setTipePendingin(String tipePendingin) {
        this.tipePendingin = tipePendingin;
    }

    public void setKecepatanRpms(int kecepatanRpms) {
        this.kecepatanRpms = kecepatanRpms;
    }

    // Method info
    public String info() {
        return "Cooling System: " + tipePendingin + " " + kecepatanRpms + " RPM (Merk: " + getMerk() + ")";
    }
}

File: Python/CoolingSystem.py
from Komponen import Komponen


class CoolingSystem(Komponen):
    def __init__(self, merk="", nama="", tipe_pendingin="", kecepatan_rpms=0):
        super().__init__(merk, nama)
        self.tipe_pendingin = tipe_pendingin
        self.kecepatan_rpms = kecepatan_rpms

    # Getter
    def get_tipe_pendingin(self):
        return self.tipe_pendingin

    def get_kecepatan_rpms(self):
        return self.kecepatan_rpms

    # Setter
    def set_tipe_pendingin(self, tipe_pendingin):
        self.tipe_pendingin = tipe_pendingin

    def set_kecepatan_rpms(self, kecepatan_rpms):
        self.kecepatan_rpms = kecepatan_rpms

    # Method info
    def info(self):
        return f"Cooling System: {self.tipe_pendingin} {self.kecepatan_rpms} RPM (Merk: {self.get_merk()})"